src/SDK/ConfigBuilder.php

```diff
@@ -0,0 +1,199 @@
+        'OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT' => 'span.limits.attribute_value_length',
+        'OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT' => 'span.limits.attribute_count',
+        'OTEL_SPAN_EVENT_COUNT_LIMIT' => 'span.limits.event_count',
+        'OTEL_SPAN_LINK_COUNT_LIMIT' => 'span.limits.link_count',
+        'OTEL_EVENT_ATTRIBUTE_COUNT_LIMIT' => 'span.limits.attribute_per_event',
+        'OTEL_LINK_ATTRIBUTE_COUNT_LIMIT' => 'span.limits.attribute_per_link',
+        'OTEL_EXPORTER_ZIPKIN_ENDPOINT' => 'trace.exporters.zipkin.endpoint',
+        'OTEL_EXPORTER_ZIPKIN_TIMEOUT' => 'trace.exporters.zipkin.timeout',
+        'OTEL_EXPORTER_OTLP_ENDPOINT' => 'trace.exporters.otlp.endpoint',
+        'OTEL_EXPORTER_OTLP_TRACES_ENDPOINT' => 'trace.exporters.otlp.endpoint',
+        'OTEL_EXPORTER_OTLP_INSECURE' => 'trace.exporters.otlp.insecure',
+        'OTEL_EXPORTER_OTLP_SPAN_INSECURE' => 'trace.exporters.otlp.insecure',
+        'OTEL_EXPORTER_OTLP_CERTIFICATE' => 'trace.exporters.otlp.certificate_file',
+        'OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE' => 'trace.exporters.otlp.certificate_file',
+        'OTEL_EXPORTER_OTLP_HEADERS' => 'trace.exporters.otlp.headers',
+        'OTEL_EXPORTER_OTLP_TRACES_HEADERS' => 'trace.exporters.otlp.headers',
+        'OTEL_EXPORTER_OTLP_COMPRESSION' => 'trace.exporters.otlp.compression',
+        'OTEL_EXPORTER_OTLP_TRACES_COMPRESSION' => 'trace.exporters.otlp.compression',
+        'OTEL_EXPORTER_OTLP_TIMEOUT' => 'trace.exporters.otlp.timeout',
+        'OTEL_EXPORTER_OTLP_TRACES_TIMEOUT' => 'trace.exporters.otlp.timeout',
+        'OTEL_EXPORTER_OTLP_PROTOCOL' => 'trace.exporters.otlp.protocol',
+        'OTEL_EXPORTER_OTLP_TRACES_PROTOCOL' => 'trace.exporters.otlp.protocol',
+        'OTEL_EXPORTER_JAEGER_AGENT_HOST' => 'trace.exporters.jaeger.agent_host',
+        'OTEL_EXPORTER_JAEGER_AGENT_PORT' => 'trace.exporters.jaeger.agent_port',
+        'OTEL_EXPORTER_JAEGER_ENDPOINT' => 'trace.exporters.jaeger.endpoint',
+        'OTEL_EXPORTER_JAEGER_TIMEOUT' => 'trace.exporters.jaeger.timeout',
+        'OTEL_EXPORTER_JAEGER_USER' => 'trace.exporters.jaeger.user',
+        'OTEL_EXPORTER_JAEGER_PASSWORD' => 'trace.exporters.jaeger.password',
+        'OTEL_PROPAGATORS' => 'propagators',
+        'OTEL_TRACES_EXPORTER' => 'trace.exporter',
+        'OTEL_METRICS_EXPORTER' => 'metrics.exporter',
+        'OTEL_PHP_TRACES_PROCESSOR' => 'trace.processor',
+    ];
+
+    public function build(array $data = []): object
+    {
+        //@phan-ignore PhanUndeclaredClassMethod
+        $schema = Expect::structure([
+            'log' => Expect::structure([
+                'stream' => Expect::string('php://stdout'),
+                'level' => Expect::string('info'),
+            ]),
+            'propagators' => Expect::string('tracecontext,baggage'),
+            'service' => Expect::structure([
+                'name' => Expect::string(),
```

Comment:
we could add all defaults here, and remove from the various classes that look up env vars